addons/website/static/src/js/editor/wysiwyg_multizone.js

```diff
@@ -574,6 +583,45 @@ var WysiwygMultizone = Wysiwyg.extend({
         this._focusedNode = node;
         this._super.apply(this, arguments);
     },
+    /**
+     * @override
+     */
+    _onShowDropdown: function (ev) {
+        $(ev.currentTarget).find('.dropdown-menu').addClass('show');
+        ev.preventDefault();
+        this.snippets.trigger('toggle_mega_menu_snippets', true);
+    },
+    /**
+     * Prevent dropdown closing when a contenteditable children is focused
+     *
+     * @override
+     */
+    _onHideDropdown: function (ev) {
+        if (ev.originalEvent &&
+                $(ev.target).has(ev.originalEvent.target).length &&
+                $(ev.originalEvent.target).is('[contenteditable]')) {
+            ev.preventDefault();
+        }
+        if (this.dropdownClickedElement) {
+            ev.preventDefault();
+        }
+        delete this.dropdownClickedElement;
```

Comment:
While it's much cleaner with your method, it still did not work as expected (dropdown still closing when clicking on overlay -> appears when an element is focused in edit mode) so what I also did is to avoid listening event when the user clicked on an overlay. It should do the trick.